File: 백준/Silver/10866. 덱/덱.py
import sys; input = lambda:sys.stdin.readline().rstrip()
from collections import deque

dq = deque()
n = int(input())
for _ in range(n):
    args = input().split()
    if args[0] == 'push_front':
        dq.appendleft(int(args[1]))
    elif args[0] == 'push_back':
        dq.append(int(args[1]))
    elif args[0] == 'pop_front':
        if len(dq) == 0:
            print(-1)
        else:
            print(dq.popleft())
    elif args[0] == 'pop_back':
        if len(dq) == 0:
            print(-1)
        else:
            print(dq.pop())
    elif args[0] == 'size':
        print(len(dq))
    elif args[0] == 'empty':
        if len(dq) == 0:
            print(1)
        else:
            print(0)
    elif args[0] == 'front':
        if len(dq) == 0:
            print(-1)
        else:
            print(dq[0])
    elif args[0] == 'back':
        if len(dq) == 0:
            print(-1)
        else:
            print(dq[len(dq)-1])
    else:
        print("wrong command")

File: 백준/Silver/10866. 덱/덱.cc
#include <iostream>
#include <deque>

using namespace std;

int main(){
    cin.tie(nullptr);   ios::sync_with_stdio(false);
    int n,k;
    cin>>n;
    deque<int> deq;
    while(n--){
        string command;
        cin>>command;
        if(!command.compare("push_front")){
            cin>>k;
            deq.push_front(k);
        }
        else if(!command.compare("push_back")){
            cin>>k;
            deq.push_back(k);
        }
        else if(!command.compare("pop_front")){
            if(deq.empty()){
                cout<<"-1\n";
                continue;
            }
            cout<<deq.front()<<'\n';
            deq.pop_front();
        }
        else if(!command.compare("pop_back")){
            if(deq.empty()){
                cout<<"-1\n";
                continue;
            }
            cout<<deq.back()<<'\n';
            deq.pop_back();
        }
        else if(!command.compare("size")){
            cout<<deq.size()<<'\n';
        }
        else if(!command.compare("empty")){
            if(deq.empty()) cout<<"1\n";
            else            cout<<"0\n";
        }
        else if(!command.compare("front")){
            if(deq.empty()){
                cout<<"-1\n";
                continue;
            }
            cout<<deq.front()<<'\n';
        }
        else if(!command.compare("back")){
            if(deq.empty()){
                cout<<"-1\n";
                continue;
            }
            cout<<deq.back()<<'\n';
        }
    }
}